frame/system/src/lib.rs

```diff
@@ -912,6 +960,11 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Deposits a log and ensures it matches the block's log data.
+	/// 
+	/// # <weight>
+	/// - `O(D)` where `D` length of `Digest`
+	/// - 1 storage mutation (codec `O(D)`).
+	/// # </weight>
 	pub fn deposit_log(item: DigestItemOf<T>) {
 		let mut l = <Digest<T>>::get();
 		l.push(item);
```

Comment:
We may be able to optimize this with `append` api. Do you want to try?

https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageValue.html#tymethod.append

Ideally there are no trait issues

Reply:
I was wondering the same. I don't think I want to introduce another PR, but I can include a commit in this one.

Reply:
`Digest` is not a vec, but a wrapper around it. `EncodeAppend` cannot be derived, but needs to be implemented. It's simple, but out of scope for now.